src/Listener.js

```diff
@@ -0,0 +1,63 @@
+const desktopIdle = require('desktop-idle');
+
+// We want the status of the user to be accurate.
+// To ensure this we need to frequently check the users status
+const DEFAULT_POLLING_TIMEOUT = 1000;
+
+class Listener {
+  constructor(id, timeToAway, callback) {
+    this.id = id;
+    this.timeToAway = timeToAway;
+    this.timeoutId = null;
+    this.callback = callback;
+    this.isAway = false;
+    this.shouldListen = true;
+  }
+
+  checkIsAway() {
+    let idleSeconds;
+
+    try {
+      idleSeconds = desktopIdle.getIdleTime();
+    } catch (error) {
+      this.callback(null, error);
+      this.scheduleCheckIsAway();
+      return;
+    }
+
+    const isAway = idleSeconds >= this.timeToAway;
+
+    if (!this.isAway && isAway) {
```

Comment:
Using `this.isAway` and `isAway` kind of makes this a bit difficult to reason about (similar names, comparing past and present time flags). How about we cache `this.isAway` in a variable with a more descriptive name:

```js
const isAway = idleSeconds >= this.timeToAway;
const wasAway = this.isAway

if (!wasAway && isAway) {
  // ...
}
```